fe2-android/app/src/main/java/com/github/dedis/popstellar/model/network/method/message/data/federation/Challenge.kt

```diff
@@ -19,6 +20,7 @@ class Challenge
 
   init {
     this.validUntil = max(0L, validUntil)
+    MessageValidator.verify().isNotEmptyBase64(value, "challenge").validFutureTimes(this.validUntil)
```

Comment:
By checking the protocol, it seems that the `value ` argument has some guidelines that should probably be checked (the regex pattern and encoding, the 32 byte length). Also the encoding seems to be Hex and not base64, I am not sure it this changes something about it.

<details>
<summary> json protocol </summary>

```
// ../protocol/query/method/message/data/dataFederationChallenge.json

{
    "$schema": "http://json-schema.org/draft-07/schema#",
    "$id": "https://raw.githubusercontent.com/dedis/popstellar/master/protocol/query/method/message/data/dataFederationChallenge.json",
    "description": "Challenge object in the context of federation authentication",
    "type": "object",
    "properties": {
        "object": {
            "const": "federation"
        },
        "action": {
            "const": "challenge"
        },
        "value": {
            "type": "string",
            "contentEncoding": "hex",
            "pattern": "^[0-9a-fA-F]{64}$",
            "$comment": "A 32 bytes array encoded in hexadecimal"
        },
        "valid_until": {
            "type": "integer",
            "description": "[Timestamp] of the expiration time",
            "minimum": 0
        }
    },
    "additionalProperties": false,
    "required": [
        "object",
        "action",
        "value",
        "valid_until"
    ]
}
```


</details


Also, why is `this.value` never assigned? I might just not have catch the thing

Reply:
Ooh true, good catch ! I thought it was base64. However the challenges are not always hex, I need to check that with the team. For now, I'll check that it is not empty, otherwise it would break everything. Oh and this.value is set when constructing the challenge